internal/function_window_option.go

```diff
@@ -314,7 +317,11 @@ func parseWindowOptions(args ...Value) ([]Value, *WindowFuncStatus, error) {
 		case WindowFuncOptionEnd:
 			opt.End = v.Value.(*WindowBoundary)
 		case WindowFuncOptionPartition:
-			opt.Partitions = append(opt.Partitions, v.Value.(Value))
+			if v.Value != nil {
+				opt.Partitions = append(opt.Partitions, v.Value.(Value))
+			} else {
+				opt.Partitions = append(opt.Partitions, windowNilPartitionValue)
```

Comment:
Can't we just use `nil` instead of `windowNilPartitionValue` ?

Reply:
What would `strings.Join("_", []string{"a", nil, "c"}` evaluate to? 

The intention is that windowed rows that share common `nil` columns are partitioned together.

```sql
WITH toks as (
  SELECT 'a' as prop1, null as prop2, 'aa' as prop3
  UNION ALL SELECT 'a' as prop1, 'aa' as prop2, null as prop3
)
SELECT prop1, ROW_NUMBER() over (PARTITION BY prop2, prop3) AS rn FROM toks
```

Expected:
| prop1 | rn |
| -- | -- |
|  a | 1 |
| a | 1 |

Reply:
A guard expression against nil must be inserted before the join process.

```
func (s *WindowFuncStatus) Partition() (string, error) {
	partitions := make([]string, 0, len(s.Partitions))
	for _, p := range s.Partitions {
                if p == nil {
                  continue
                }
		text, err := p.ToString()
		if err != nil {
			return "", err
		}
		partitions = append(partitions, text)
	}
	return strings.Join(partitions, "_"), nil
}
```

Reply:
In the query above, wouldn't that result in the following?

| prop1 | rn |
| -- | -- |
|  a | 1 |
| a | 2 |

Reply:
Ah, I see. Then it looks like we should add constant `NullValue` structure.

Example.

- value.go

```go
type NullValue struct {
  zeroValue Value
}
```

`NullValue` should have the original type information, so it should either have a zero value or `reflect.Type` .
If `NULL` value is integer type, initialize as follows

```go
&NullValue{zeroValue: IntValue(0)}
```

Reply:
I will try to do this as I think it is the right approach, too.

Reply:
Thank you !! Generics seems to be a better choice.  😃 

```go
type NullValue[T Value] struct {
  zeroValue T
}
```